flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/AdaptiveBatchScheduler.java

```diff
@@ -179,6 +183,7 @@ protected void onTaskFinished(final Execution execution, final IOMetrics ioMetri
         checkNotNull(ioMetrics);
         updateResultPartitionBytesMetrics(ioMetrics.getResultPartitionBytes());
         initializeVerticesIfPossible();
+        tryEnrichInputNumBytesForExecutionVertexInputInfos();
```

Comment:
This gradually builds up the input bytes for an execution vertex, even if the execution vertex is already running.
However, a changing input bytes is not ideal to slow task detector at the moment, because a detected slow task may not be detected as slow later, if a new input is finished and the input bytes number is accumulated. This may result in some inconsistency.

I'd like to avoid it, but instead compute the input bytes of execution vertices when before scheduling them, i.e. by overriding the `allocateSlotsAndDeploy` method.

Reply:
Yes, it's better, done with your suggested way.